git/operations.go

```diff
@@ -276,35 +276,94 @@ func changed(ctx context.Context, workingDir, ref string, subPaths []string) ([]
 	return splitList(out.String()), nil
 }
 
-func execGitCmd(ctx context.Context, args []string, config gitCmdConfig) error {
-	if trace {
-		print("TRACE: git")
-		for _, arg := range args {
-			print(` "`, arg, `"`)
+// Whilst debugging or developing, you may wish to filter certain git commands out of the logs.
+func exemptedTraceGitCommand(command string) bool {
+	exemptedCommands := []string{
+		// To filter out a certain git subcommand add it here, e.g.:
+		// "config",
+	}
+	for _, exemptedCommand := range exemptedCommands {
+		if exemptedCommand == command {
+			return true
+		}
+	}
+	return false
+}
+
+// traceGitCommand returns a log line that can be useful when debugging and developing git activity
+func traceGitCommand(args []string, config gitCmdConfig, stdout string, stderr string) string {
+	if exemptedTraceGitCommand(args[0]) {
+		return ""
+	}
+
+	prepare := func(input string) string {
+		output := strings.Trim(input, "\x00")
+		scanner := bufio.NewScanner(strings.NewReader(output))
```

Comment:
`return strings.Replace(output, "\n", "\\n", -1)`

(the `-1` means replace all; there is a ReplaceAll, but it seems to be in a more recent Go stdlib)

Reply:
I had used a scanner because on a call when I presented this issue to you (the issue of what method to opt for for replacing the newlines) you had said to use a scanner and scan by line.

I like the strings.Replace option, thanks :)

Since we're taking the "use the standard lib" approach more, I also replaced the other for loop to `strings.TrimSuffix(output, "\n")`.